tests/io/test_h5io.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+
+from atacworks.io import h5io
+import h5py
+import numpy as np
+import pytest
+import os
+
+
+@pytest.mark.cpu
+def test_dict_to_h5(tmpdir):
+    """Test dict to h5."""
+
+    input_dict = {"features": np.array([0, 1, 2, 3]),
+                  "scores": np.array([1, 2, 1, 1])}
+    h5file = os.path.join(tmpdir, "h5file.h5")
+    h5io.dict_to_h5(input_dict, h5file)
+    with h5py.File(h5file, 'r') as f:
+        for key in input_dict.keys():
+            data = np.array(f[key])
+            assert data.all() == input_dict[key].all()
+
+    # Append to existing dataset and test.
+    input_dict = {"features": np.array([0, 1, 2, 3]),
+                  "scores": np.array([1, 2, 1, 1])}
+    new_dict = {"features": np.array([0, 1, 2, 3, 4, 5, 6])}
+    input_dict.update(new_dict)
+    h5io.dict_to_h5(new_dict, h5file, create_new=False)
+    with h5py.File(h5file, 'r') as f:
+        for key in input_dict.keys():
+            data = np.array(f[key])
+            assert data.all() == input_dict[key].all()
+
+
+@pytest.mark.cpu
+def test_h5_to_array(tmpdir):
+    """Test h5 to array."""
+
+    input_dict = {"features": np.array([0, 1, 2, 3]),
+                  "scores": np.array([1, 2, 1, 1])}
+    h5file = os.path.join(tmpdir, "h5file.h5")
+    with h5py.File(h5file) as f:
+        for key in input_dict.keys():
+            f.create_dataset(key, data=input_dict[key])
+    data = h5io.h5_to_array(h5file, "features", None)
+    assert data.all() == input_dict["features"].flatten().all()
```

Comment:
neat! can you update this to a loop that goes over all the keys in the original `input_dict` and runs the test for each?

Reply:
done.